services/clojars/clojars-downloads.service.js

```diff
@@ -0,0 +1,67 @@
+'use strict'
+
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+const { metric } = require('../../lib/text-formatters')
+const { NotFound } = require('../errors')
+const { nonNegativeInteger } = require('../validators')
+const { downloadCount: downloadsColor } = require('../../lib/color-formatters')
+
+const clojarsSchema = Joi.object({
+  downloads: nonNegativeInteger,
+}).required()
+
+module.exports = class Clojars extends BaseJsonService {
+  async fetch({ clojar }) {
+    const url = `https://clojars.org/api/artifacts/${clojar}`
+    return this._requestJson({
+      url,
+      schema: clojarsSchema,
+    })
+  }
+
+  static render({ downloads }) {
+    return {
+      label: 'downloads',
+      message: metric(downloads),
+      color: downloadsColor(downloads),
+    }
+  }
+
+  async handle({ clojar }) {
+    const json = await this.fetch({ clojar })
+
+    if (json.downloads === undefined) {
+      throw new NotFound()
+    }
```

Comment:
This check can be removed. The typical "not found" case is a 404 (which gets handled by `_requestJson()` )

```sh
$ curl -I "https://clojars.org/api/artifacts/not-a-package"
HTTP/1.1 404 Not Found
```

If we've got a `200 OK` response and `downloads` is `undefined`, it will be rejected by the schema because `nonNegativeInteger` is `required()`: https://github.com/badges/shields/blob/510491f37644432740548afd8cdbac7e1978b59f/services/validators.js#L6-L9